### DISCUSSION TITLE
Switching between env versions

### DISCUSSION THREAD

**Discussion by fristyr** [goatandsheep/react-native-dotenv#229] [Q&A]
Hi

I have 3 env files in my project

.env.production
.env.development
.env.test

I was trying to follow the guide from page
but after I try a script

    "start:production": "SET NODE_ENV=development & expo start",

No errors, expo starts, but then I still see the development env variables

**Comment by mariomurrent-softwaresolutions** [comment]
Hi you need to add --clear 

`start:production": "SET NODE_ENV=development & expo start --clear`

I just did not manage to get it working with `expo build`

**Reply by fristyr** [reply]
> Hi you need to add --clear
> 
> `start:production": "SET NODE_ENV=development & expo start --clear`
> 
> I just did not manage to get it working with `expo build`

Are you sure?
Does not work for me

**Reply by mariomurrent-softwaresolutions** [reply]
Yes this is working for me. Just without the SET at the beginning

**Comment by goatandsheep** [comment]
Hi @fristyr have you tried using `expo --no-dev start`? https://docs.expo.dev/workflow/development-mode/

**Comment by fristyr** [comment]
@goatandsheep 
For some reason none of mentioned above help.
Maybe I am doing something wrong, so here is what I have 

.package.json
```javascript
{
  "main": "node_modules/expo/AppEntry.js",
  "scripts": {
    "start": "expo start",
    
    "start:production": "SET NODE_ENV=production & expo start",
    "start:development": "SET NODE_ENV=development & expo start",
    "start:test": "SET NODE_ENV=test & expo start",

    "android": "expo start --android",
    "ios": "expo start --ios",
    "web": "expo start --web",
    "eject": "expo eject",
    "test": "jest --watchAll"
  },
  "jest": {
    "preset": "jest-expo"
  },
  "dependencies": {
     ...
     "react-native-dotenv": "^3.3.1",
  },
  "devDependencies": {
    ...
    "@types/react-native-dotenv": "^0.2.0",
  },
  "private": true
}

```

babel.config.js
```javascript
module.exports = function (api) {
  api.cache(true);
  return {
    presets: ['babel-preset-expo'],
    plugins: [
      'react-native-reanimated/plugin',
      [
        'module-resolver',
        {
          alias: {
            app: './app',
          },
        },
      ],
      [
        'module:react-native-dotenv',
        {
          moduleName: '@env',
          path: './app/config/env',
          safe: false,
          allowUndefined: true,
        },
      ],
    ],
  };
};

```

inside of the app folder I have next
app/config/env/
which includes inside 4 files
.env
.env.development
.env.production
.env.test

**Reply by goatandsheep** [reply]
`api.cache(false)`

**Reply by goatandsheep** [reply]
also, react-native-dotenv should be devDependency only
